platform/darwin/src/MGLStyle.mm

```diff
@@ -350,7 +358,11 @@ - (void)addLayer:(MGLStyleLayer *)layer
          layer];
     }
     [self willChangeValueForKey:@"layers"];
-    [layer addToMapView:self.mapView];
+    try {
+        [layer addToMapView:self.mapView];
+    } catch (std::runtime_error & err) {
```

Comment:
We're catching a generic exception here, but always forward a `MGLRedundantLayerIdentifierException`. What we should do long term is to convert all of the generic `std::runtime_error` into individual exception types, and to exception-specific catches that we can then forward.